File: abc350a/abc350a.ml
open Printf
open Scanf

let solve s =
  let n = int_of_string @@ String.sub s 3 3 in
  if n = 0 then "No"
  else if n = 316 then "No"
  else if n >= 350 then "No"
  else "Yes"

let () =
  scanf "%s " solve |> printf "%s\n"

File: abc350a/abc350a.hs
import Data.Bool (bool)

main :: IO ()
main = solve <$> getLine >>= putStrLn

solve :: String -> String
solve = bool "Yes" "No" . f . read . drop 3
  where
    f :: Int -> Bool
    f x = x == 0 || x == 316 || x >= 350
